client/my-sites/marketplace/constants.ts

```diff
@@ -1,26 +1,123 @@
 /**
  * External dependencies
  */
-import { YOAST_SEO } from '@automattic/calypso-products';
+import { YOAST_PREMIUM, YOAST_FREE } from '@automattic/calypso-products';
 import debugFactory from 'debug';
 
 export const marketplaceDebugger = debugFactory( 'marketplace-debugger' );
 
 export const MARKETPLACE_FLOW_ID = 'marketplace_flow';
 export const ANALYTICS_UI_LOCATION_MARKETPLACE_DOMAIN_SELECTION = 'marketplace_domain_selection';
 
-// Marketplace plugin - product relationship mapped by SLUG
+/**
+ * Internal dependencies
+ */
+import {
+	IProductDefinition,
+	IProductCollection,
+	IProductGroupCollection,
+	YOAST,
+} from 'calypso/my-sites/marketplace/types';
+
+// ********** This section to be refactored out with new definition below **********
 export default interface PluginProductMappingInterface {
 	readonly 'wordpress-seo': string;
 	readonly 'wordpress-seo-premium': string;
 }
 
-// TODO: Refactor to a product indexed map of plugins
 export const PLUGIN_PRODUCT_MAP: PluginProductMappingInterface = {
-	'wordpress-seo': YOAST_SEO,
-	'wordpress-seo-premium': YOAST_SEO,
+	'wordpress-seo': YOAST_PREMIUM,
+	'wordpress-seo-premium': YOAST_PREMIUM,
 };
 
 export function getProductSlug( pluginSlug: keyof PluginProductMappingInterface ): string {
 	return PLUGIN_PRODUCT_MAP[ pluginSlug ];
 }
+// ************************************************************************************
+
+// TODO: Integrate this data structure with marketplace code
+export const productGroups: IProductGroupCollection = {
+	[ YOAST ]: {
+		products: {
+			[ YOAST_PREMIUM ]: {
+				defaultPluginSlug: 'wordpress-seo',
+				pluginsToBeInstalled: [ 'wordpress-seo-premium', 'wordpress-seo', 'yoast-test-helper' ],
```

Comment:
Nit: do these plugins have to be in any particular order? Is it possible that `wordpress-seo` is a requirement for `wordpress-seo-premium`, and in that case, it would have to be installed first?

Reply:
No order here

Reply:
👍